### PR TITLE
doc: improve document for api plugin

### LINKED ISSUE #422: More documentation for API-plugin and user custom component
**What would you like to be added**:
We need more documentation for the [API-plugin](https://mosn.io/layotto/#/en/start/api_plugin/helloworld) and [user custom component](https://github.com/mosn/layotto/issues/416) features. Currently there are only quickstart demos and design doc.

**Why is this needed**:
Because users come to ask relevant questions

### PR DESCRIPTION
**Pull request by seeflood**
**What this PR does**:
Improve document for api plugin.
Explain how to use it

**Which issue(s) this PR fixes**:

Fixes #422

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cmd/layotto/main.go [hunk 1]
```diff
@@ -205,7 +205,7 @@ func NewRuntimeGrpcServer(data json.RawMessage, opts ...grpc.ServerOption) (mgrp
 			}
 			return actuator.NewGrpcServerWithActuator(server)
 		}),
-		// register your grpc API here
+		// register your GrpcAPI here
```
Comment (zhenjunMa) [inline cmd/layotto/main.go:208]: 一般大家都写gRPC

**Review by zhenjunMa — APPROVED** [review]

**Review by stulzq — DISMISSED** [review]

File: cmd/layotto/main.go (hunk 1 above)
Comment (seeflood) [reply cmd/layotto/main.go:208]: 尴尬了，那个interface起的名字叫 `GrpcAPI` ...

Comment (seeflood) [reply cmd/layotto/main.go:208]: 那我去改回来吧

Comment (seeflood) [reply cmd/layotto/main.go]: fixed

**Review by zhenjunMa — APPROVED** [review]

**Review by stulzq — APPROVED** [review]

**Pull request merged**